definitions/npm/async_v1.x.x/flow_>=v0.28.x/async_v1.x.x.js

```diff
@@ -0,0 +1,77 @@
+// @flow
+
+declare module 'async' {
+  declare type Collection<TItem> = Iterable<TItem> | { [key: string]: TItem };
+
+  declare type IterateeCallback<TError, TResult> = (err?: TError, result?: TResult) => void;
+  declare type Callback<TError, TResult> = (err?: TError, result?: TResult) => mixed;
+
+  // Collections
+  declare function each<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (item: TItem, callback: IterateeCallback<TError, void>) => mixed,
+    callback?: Callback<TError, void>
+  ): void;
+  declare function forEachOf<TItem, TError>(
+    collection: Collection<TItem>,
+    iteratee: (value: TItem, key: string | number, callback: IterateeCallback<TError, void>) => mixed,
```

Comment:
It's almost not part of the Iterable type though. For non-objects, they key is an index that is incremented separately.

So if I write the `Collection` type like this:

```js
declare type Collection<TKey, TValue> = Iterable<TValue> | { [key: TKey]: TValue };
```

And the method like this:

```js
declare function forEachOf<TKey, TValue, TError>(
  collection: Collection<TKey, TValue>,
  iteratee: (value: TValue, key: TKey | number, ...) => mixed,
  ...
```

What I expect is that it will use the type of `TKey` if it is an object, and it will use `number` if it is an iterable.

However, [this is what happens when I use it that way](https://flowtype.org/try/#0PQKgBAAgZgNg9gdzCYAoVAXAngBwKZgDCcMMeAxhgJZwB2APACoDSeWANGIwGoCGMAVzwA+MAF4wASQx4ATrwBGZJn0EiwAHzABvMAG0A1mwBcXVlgC6pnvyFgAvgG502fFJnyZeQvxgLe5AZMAKKysnCynIwASngAzgIwGKISABRysgD81qHhkWCy8YkY2VyxCUkAlOKiAG5wVAAmzq4EPqT+gSFhEVHlxSlg6WGljLm9BUVJo-1VNWAAtlQAHnjN6FACtJQ0tGBQEcEBABYA8lBM5lGqQlHjssKpqGBg5CRkO3SmxKQU1HSXNjXWwidjPMBUDy8LymVK1EHWG54ThGLDWcyaMC0AQLBRyTjkXydAymaRyaF4bxEgJBMY9fL1JrCapiURLVaNMEvQkdGmldp+GndPKcRmNYSoSqmMU6ewbQ4nc6pXS8UwAcgAgmqHJw4SCUSYsTi8fkeYLAizRNp7JVnAdZEdyGcoMqwKqwJrtfZdfC1Aa0WA4hhZFRaABzAnUi3za221BAA).